docs/journal/02-exploring-recent-events.md

```diff
@@ -1,7 +1,122 @@
+# Journal 2: Ad Hoc Exploration
 
+We use the term "ad hoc exploration" for scenarios where the user wants to query event data on the fly. 
+ 
+For example, you might use ad hoc exploration for incident investigation. Some failure occurs in the system, and you want to know if the device readings can pinpoint the cause. A typical query might look for an average value over a time period, or look for spikes in the data.    
 
-# Lessons Learned - What Fabrikam discovered
 
-TODO List the insights, points of confusion, and challenges that we encountered during this step of the journey
+## Common requirements
+
+- Queries are ad hoc. You may not know in advance what queries are needed.
+- Queries must execute reasonably quickly, so the user can explore the data. This differs from running a batch query at some fixed interval.
+- Queries typically involve a time range: “What was the average temperature during this 20-minute window?” 
+- Queries might need relational data: “What was the average temperature across all sensors in building 25?” 
+- Relational data might be stored separately from event data; for example, a device registry that maps device IDs to buildings.
+- Ad hoc exploration *tends* to target more recent data, rather than the entire set of historical data. The exact definition of “recent” depends on your particular scenario. It might mean the past 24 hours, the past week, or the past month.
+- The system must aggregate results into a meaningful view. IoT solutions can generate a *lot* of data, even across a relatively small time slice, so you need a way for a human to understand the results.
+- Ad hoc exploration is generally "user initiated," meaning the user submits a query (or perhaps logs into a dashboard), versus having the system detect a condition and notify the user.
+
+> We don't include critical alerts under ad hoc exploration. Ad hoc queries should execute quickly, but not necessarily in real time. (See the [real-time processing][real-time-milestone] milestone.) 
```

Comment:
We might want to differentiate from "Hot" data such as within 5 min of the event occurance. Ad hoc exploration may not be able to access data this young: indexing and latency time could be more than 5 min.